api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto

```diff
@@ -16,30 +16,28 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Upstream HTTP/1.1 Proxy]
 // [#extension: envoy.transport_sockets.http_11_proxy]
 
-// The HTTP/1.1 proxy transport socket opens an upstream connection to a specified proxy address
-// rather than the target host's address. If this transport socket is configured and proxy
-// information is configured, then:
+// HTTP/1.1 proxy transport socket establishes an upstream connection to a proxy address
+// instead of the target host's address. This behavior is triggered when the transport
+// socket is configured and proxy information is provided.
 //
-// * Upstream connections to the proxy address will have a raw HTTP/1.1 CONNECT header prefaced to
-//   the payload, and 200 response stripped (if less than 200 bytes).
-// * Plaintext HTTP/1.1 connections will be sent with a fully qualified URL.
+// Behavior when proxying:
+// =======================
+// * Upstream connections to the proxy include a raw HTTP/1.1 CONNECT header before
```

Comment:
Suggest alternate wording here:

"""
When an upstream connection is established, instead of connecting directly to the endpoint address, the client will connect to the specified proxy address, send an HTTP/1.1 CONNECT request indicating the endpoint address, and process the response.  If the response has HTTP status 200, the connection will be passed down to the underlying transport socket.
"""

I assume that we always use the endpoint address in the CONNECT request, not the hostname associated with the endpoint (if any).

Reply:
> I assume that we always use the endpoint address in the CONNECT request, not the hostname associated with the endpoint (if any).

Actually, it depends on how the proxy address is configured... If done via filter state, it will be the hostname. I think I was trying to convey this when I butchered the second bullet point.

Given that it's an Envoy implementation detail, I'll leave that part out and add these details to the Envoy docs.